lib/mayaUsd/render/MaterialXGenOgsXml/GlslOcioNodeImpl.cpp

```diff
@@ -174,53 +175,81 @@ GlslOcioNodeImpl::registerOCIOFragment(const std::string& fragName, DocumentPtr
         return {};
     }
 
+    OcioData ocioData;
+
     // Validate that the fragment structure is 100% as expected. We need the properties, values,
     // outputs, and GLSL source to proceed:
     auto fragment = doc.child(TAG_FRAGMENT);
     if (!fragment) {
         return {};
     }
 
+    ocioData.fragName = fragment.attribute(ATTR_NAME).as_string();
+    if (ocioData.fragName.empty()) {
```

Comment:
Parsing everything in this function so we can release the XML document at the end.